core/src/utils/account_id.rs

```diff
@@ -100,16 +105,16 @@ impl AccountId32 {
 }
 
 /// An error obtained from trying to interpret an SS58 encoded string into an AccountId32
-#[derive(thiserror::Error, Clone, Copy, Eq, PartialEq, Debug)]
+#[derive(Clone, Copy, Eq, PartialEq, Debug, Display)]
 #[allow(missing_docs)]
 pub enum FromSs58Error {
-    #[error("Base 58 requirement is violated")]
+    #[display(fmt = "Base 58 requirement is violated")]
     BadBase58,
-    #[error("Length is bad")]
+    #[display(fmt = "Length is bad")]
     BadLength,
-    #[error("Invalid checksum")]
+    #[display(fmt = "Invalid checksum")]
     InvalidChecksum,
-    #[error("Invalid SS58 prefix byte.")]
+    #[display(fmt = "Invalid SS58 prefix byte.")]
```

Comment:
Perhaps worth an `if std, impl std::error::Error` just to be consistent with how we ususally treat errors?